aten/src/ATen/native/TensorAdvancedIndexing.cpp

```diff
@@ -830,6 +830,17 @@ TORCH_IMPL_FUNC(index_add_cpu_out)
 
   auto index_contig = index.contiguous();
 
+  if (source.size(dim) != index.size(0)) {
```

Comment:
This doesn't seem right.

Reply:
Thank you so much for your review, apologies for getting to this so late.

First, I'll definitely add a test first before making changes -- when you say adding a sample of the relevant size do you mean something that should reproduce the intended error? Just want to make sure I'm clear on this before going ahead and doing it.

Reply:
Thank you so much for your review, apologies for getting to this so late.

First, I'll definitely add a test first before making changes -- when you say adding a sample of the relevant size, do you mean a method invocation that should reproduce the intended error? Just want to make sure I'm clear on this before going ahead and doing it.

For the dimensionality check, isn't this the correct way to check "source.shape[dim] == len(index)"? Since index must be a vector, can't we use .size(0) to refer to its length? That being said, many of the errors seem to be that the "dimension specified as 0 but tensor has no dimensions" -- maybe I need an error check that index isn't empty? I'm sure I'm missing something here, any further elaboration would be extremely helpful.

Finally, on the vectorization front. I'm thinking the following code might be a better way of doing this operation (drawing inspiration from the implementation of index_copy):

```
  auto self_sizes = self.sizes().vec();
  auto source_sizes = source.sizes().vec();
  self_sizes.erase(self_sizes.begin() + dim);
  source_sizes.erase(source_sizes.begin() + dim);
  if (self.sizes() != source.sizes()) {
      error;
  }
```

Please let me know what you think, and thank you again for your review!

Reply:
Index can also be a 0-dim tensor, so `index.size(0)` would not work in that case. In any case, it's better to be more explicit calling ìndex.numel(). For the rest, yes, something along those lines would make sense.

Reply:
Just fixed -- on the testing front, can we add a one-off test somewhere in common_methods_invocations, or should I put some more effort into determining how sample_inputs_index is used?

Reply:
You should add the relevant error sample to `common_methods_invocations`